lightning/src/ln/outbound_payment.rs

```diff
@@ -752,8 +1021,15 @@ impl OutboundPayments {
 				// process_onion_failure we should close that channel as it implies our
 				// next-hop is needlessly blaming us!
 				if let Some(scid) = short_channel_id {
+					outbounds.get_mut(&payment_id).map(|pmt| pmt.insert_previously_failed_scid(scid));
```

Comment:
It didn't seem to make sense to move the `payment_is_probe` call, but otherwise did this. 

I had considered this approach but thought it might be worth the extra `outbounds` lock to do some cheap checks before decoding the onion failure, but those cheap checks look pretty rare anyway